docs/source/model/signatures.rst

```diff
@@ -629,6 +629,127 @@ The same signature can be created explicitly as follows:
+helps validate that the model can run predictions successfully, making it **strongly recommended to always include an input example when logging a GenAI model**.
+
+.. note::
+    By default, fields in the input example are inferred as required. To mark a field as optional, you can provide a list of 
+    input examples where some examples contain None values for the optional fields. This behavior will be improved in future releases.
+
+Take langchain model as an example, you can log a model with signature with below steps:
+
+1. Define the langchain model in a separate python file named `langchain_model.py`:
+
+To log a langchain model successfully, **it is highly recommended to use** `models from code <../models.html#models-from-code>`_ feature.
+
+.. code-block:: python
+
+    import os
+    from operator import itemgetter
+
+    from langchain_core.output_parsers import StrOutputParser
+    from langchain_core.prompts import PromptTemplate
+    from langchain_core.runnables import RunnableLambda
+    from langchain_openai import OpenAI
+
+    import mlflow
+
+    mlflow.set_experiment("Homework Helper")
+
+    mlflow.langchain.autolog()
+
+    prompt = PromptTemplate(
+        template="You are a helpful tutor that evaluates my homework assignments and provides suggestions on areas for me to study further."
+        " Here is the question: {question} and my answer which I got wrong: {answer}",
+        input_variables=["question", "answer"],
+    )
+
+
+    def get_question(input):
+        default = "What is your name?"
+        if isinstance(input_data[0], dict):
+            return input_data[0].get("content").get("question", default)
+        return default
+
+
+    def get_answer(input):
+        default = "My name is Bobo"
+        if isinstance(input_data[0], dict):
+            return input_data[0].get("content").get("answer", default)
+        return default
+
+
+    model = OpenAI(temperature=0.95)
+
+    chain = (
+        {
+            "question": itemgetter("messages") | RunnableLambda(get_question),
+            "answer": itemgetter("messages") | RunnableLambda(get_answer),
+        }
+        | prompt
+        | model
+        | StrOutputParser()
+    )
+
+    mlflow.models.set_model(chain)
+
+2. Log the langchain model with a valid input example:
+
+.. code-block:: python
+    
+    input_example = {
+        "messages": [
+            {
+                "role": "user",
+                "content": {
+                    "question": "What is the primary function of control rods in a nuclear reactor?",
+                    "answer": "To stir the primary coolant so that the neutrons are mixed well.",
+                },
+            },
+        ]
+    }
+
+    chain_path = "langchain_model.py"
+    with mlflow.start_run():
+        model_info = mlflow.langchain.log_model(
+            lc_model=chain_path, artifact_path="model", input_example=input_example
+        )
+
```

Comment:
Can we add an example of defining Optional values? This is another point of confusion (even though it's mentioned in text above, we should show a code snippet for readers who navigate directly to code first)